src/Smidge.Core/Cache/PhysicalFileCacheFileSystem.cs

```diff
@@ -42,7 +47,7 @@ public IFileInfo GetRequiredFileInfo(string filePath)
 
             if (!fileInfo.Exists)
             {
-                throw new FileNotFoundException($"No such file exists {fileInfo.PhysicalPath ?? fileInfo.Name} (mapped from {filePath})", fileInfo.PhysicalPath ?? fileInfo.Name);
```

Comment:
I think this should be handled differently, now fileInfo is returned even when its useless (also in MemoryCacheFileSystem).

Reply:
Which is as intended, as other parts of the logic will check for `.Exists`, which is precisely what a FileInfo is for (a cheap wrapper to tell you if a file exists before attempting to open it). You also annotated the old code :)

Reply:
> Which is as intended, as other parts of the logic will check for `.Exists`, which is precisely what a FileInfo is for (a cheap wrapper to tell you if a file exists before attempting to open it). You also annotated the old code :)
> 
> Returning null here would mean a lot more code has to be touched and double-checked.

It's already documented that GetRequiredFileInfo throws: 
https://github.com/Shazwazza/Smidge/blob/994e9451b9838e74669fd47b29489509dbfe92c7/src/Smidge.Core/ISmidgeFileSystem.cs#L24

So this would be breaking anyway. 

Not all other paths check if the file exists (only SmidgeController does):
https://github.com/Shazwazza/Smidge/blob/994e9451b9838e74669fd47b29489509dbfe92c7/src/Smidge.Core/FileProcessors/PreProcessManager.cs#L79
https://github.com/Shazwazza/Smidge/blob/994e9451b9838e74669fd47b29489509dbfe92c7/src/Smidge.Core/FileProcessors/CssImportProcessor.cs#L50

This would also apply to other people using the library, now it doesn't throw anymore and you need to handle it differently.

Reply:
Remark about breaking change is fair, the other ones are easily "fixed" by reintroducing the exception right after retrieving the IFileInfo. I will push that in a second.

The two locations you pointed out are literally the only 2 places where something is done with `GetRequiredFileInfo`. They are also only called in very specific situations: `@import` in CSS and during bundle creation, both of which cannot be directly manipulated by malicious users (only malicious developers :P). That said, bundle creation would also need some love as I pointed out, but for another PR.

Reply:
> Remark about breaking change is fair, the other ones are easily "fixed" by reintroducing the exception right after retrieving the IFileInfo.

Not everyone uses Smidge as-is, some have custom implementations around it. I don't mind fixing this at all, but let's try to get consistent behavior ☺️

I think it's fine not to throw anymore; then PreProcessManager can bail out, and CssImportProcessor can continue with the others.

Alternatively, SmidgeController could swallow the exception. This would be the least breaking change.

Reply:
Yeah, i don't think we should change the behavior of this method since the meaning is in the name 'GetRequired...' ... which indicates that it will throw. 

One way to be less breaking would be to have an optional overload of this like `bool throwIfNotFound = true`

It is a binary breaking change, but all folks nowadays build their projects before deploying so I don't think it would cause any issues. 

Else I'm fine with try/catch/swallow where necessary

Reply:
@LennardF1989 just wanted to ping on this one?

Reply:
Honestly, I forgot that I did this xD I'll put it back on my list.